File: capybara/spec/login_logout_spec.rb
require "spec_helper"

describe "logging and logging out", js: true do
  Capybara.app_host = "http://localhost:8080"
  Capybara.server_host = "localhost"
  Capybara.server_port = "8080"

  it "works" do
    visit "/uaa"

    fill_in "Email", with: "marissa"
    fill_in "Password", with: "koala"

    click_on "Sign in"

    expect(page).to have_content "Where to?"

    find(".dropdown-trigger", text: "marissa").click

    click_on "Sign Out"

    expect(page).to have_content "Welcome!"
  end
end

File: casperjs/spec/loginLogoutSpec.js
var casper = require('casper').create();

casper.start("http://localhost:8080/uaa", function() {
    this.fillSelectors('form[action="/uaa/login.do"]', {
        "input[placeholder=Email]": "marissa",
        "input[placeholder=Password]": "koala"
    }, true);
});

casper.waitForText("Where to?", function() {
    this.click('.dropdown-trigger');
    this.clickLabel('Sign Out', 'a');
});

casper.waitForText("Welcome!")

casper.run();